src/reflect/scala/reflect/internal/Symbols.scala

```diff
@@ -2752,30 +2752,35 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
 
     /** String representation of symbol's definition following its name */
     final def infoString(tp: Type): String = {
-      def parents = (
-        if (settings.debug.value) parentsString(tp.parents)
-        else briefParentsString(tp.parents)
-      )
-      def isStructuralThisType = (
-        // prevents disasters like scala/bug#8158
-        owner.isInitialized && owner.isStructuralRefinement && tp == owner.tpe
-      )
-      if (isType) typeParamsString(tp) + (
-        if (isClass) " extends " + parents
-        else if (isAliasType) " = " + tp.resultType
-        else tp.resultType match {
-          case rt @ TypeBounds(_, _) => "" + rt
-          case rt                    => " <: " + rt
+      def loop(tp: Type, first: Boolean = false): String = {
+        def parents = (
+          if (settings.debug) parentsString(tp.parents)
+          else briefParentsString(tp.parents)
+          )
+        def isStructuralThisType = (
+          // prevents disasters like scala/bug#8158
```

Comment:
You have re-enabled the disaster, like those time-travellers who are forever trying to fix the future and avert the apocalypse.

`% scalac t8158/Test_2.scala error: Java heap space `

Reply:
Maybe I was looking at the wrong jenkins output.